File: src/leetcode/CoinChange2.java
package leetcode;

public class CoinChange2 {
	public static int change(int amount, int[] coins) {
		int[] dp = new int[amount+1];
		dp[0] = 1;
		for(int coin: coins) {
			for(int i = coin; i < amount+1; i++) {
				dp[i] += dp[i - coin];
			}
		}
        return dp[amount];
    }
	public static void main(String[] args) {
		int amount = 5;
		int[] coins = new int[]{1, 2, 5};
		int result = change(amount, coins);
		System.out.println(result);
	}
}

File: src/leetcode/CoinChange2.py
class Solution(object):
    def change(self, amount, coins):
        """
        :type amount: int
        :type coins: List[int]
        :rtype: int
        """
        dp = [0] * (amount + 1)
        dp[0] = 1
        for coin in coins:
            for j in range(1, amount + 1):
                if j >= coin:
                    dp[j] += dp[j - coin]
        return dp[amount]


if __name__ == "__main__":
    amount = 5
    coins = [1, 2, 5]
    solution = Solution()
    result = solution.change(amount, coins)
    print result
